### PR TITLE
[SYCL][NATIVECPU] Fix linker errors for WorkGroup collective functions

### PR DESCRIPTION
**Pull request by PietroGhg**
This PR fixes some link-time error related to WorkGroup collective functions:
* adds overloads for the `_Float16` data type and the `__spirv_GroupBroadcast` function, which leads to `undefined reference` linker errors.
* Makes it so the call to `clang-offload-deps` is not emitted by the driver: the call introduced an `llvm.used` array, which contained function pointers to the kernels which prevented eliminating those functions from the module even when they are not needed anymore (due to inlining). This lead to other `undefined reference` linker errors for WorkGroup collectives.

### CODE REVIEW

**Review by coldav — APPROVED** [review]

File: sycl/test/native_cpu/reduce_multi_tu.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      }
+    }
+  }
+
+  if (fails == 0) {
+    printf("test passed!\n");
+  } else {
+    printf("test failed!\n");
+  }
+  assert(fails == 0);
```
Comment (uwedolinsky) [inline sycl/test/native_cpu/reduce_multi_tu.cpp:115]: The assert might get compiled away so it's probably better to return `fails` to not lose the error.

**Review by mdtoguchi — APPROVED** [review]

**Comment by PietroGhg** [comment]
Hi @intel/llvm-reviewers-runtime, after https://github.com/intel/llvm/pull/15142 the ownership of `libdevice/nativecpu*` has changes, but your approval is still required for this PR, could someone take a look? Thank you

File: sycl/test/native_cpu/reduce_multi_tu.cpp (hunk 1 above)
Comment (PietroGhg) [reply sycl/test/native_cpu/reduce_multi_tu.cpp]: That's done, thank you

**Review by uditagarwal97 — APPROVED** [review]

**Review by uwedolinsky — APPROVED** [review]

**Comment by PietroGhg** [comment]
@intel/llvm-gatekeepers this looks ready to merge, thank you

**Pull request merged**